### PR TITLE
feat: iwd challenge redirect and thank you page

### PR DESCRIPTION
**Pull request by eddieferrer**
* rename eco utils to generic achievement utils
* implement thank you page and logic iwd challenge
* added iwd badge on lending stats page
* additional renaming of directories 'eco-challenge' > 'achievements'

ACK-543

Lots of changes, but most of them are renaming files to be something a little more generic. 

Summary of scope: 
* Challenge will run 3/7-3/10 (handled in achievement service)
* Criteria will be 1 woman loan in cart. (handled in achievement service)
* UI will have a matching experiment setting `iwd_challenge` which we will turn on same dates. 
* If experiment is active, redirect to special thank you page on checkout. 
* Use the achievement service to determine badge status and display badge on lending-stats page. 
* Most of this code is reusing some parts from the previous eco-challenge
* Will push achievement service this week so we can run a trial run on dev only before our actual dates

Screen Shot 2023-02-28 at 12 28 57 PM
Screen Shot 2023-02-28 at 12 28 42 PM

### CODE REVIEW

File: src/pages/Thanks/ThanksPageIwdChallenge.vue [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+						<kv-button
+							class="tw-mt-3 tw-ml-3"
+							href="/gender-equality"
+							variant="secondary"
+							style="max-width: 190px;"
+						>
+							Learn More
+						</kv-button>
+					</div>
+				</div>
```
Comment (mcstover) [inline src/pages/Thanks/ThanksPageIwdChallenge.vue:109]: It is possible to add a Portfolio link somewhere on this page like on our standard thanks page share version?

Comment (eddieferrer) [reply src/pages/Thanks/ThanksPageIwdChallenge.vue:115]: Added in new version

File: src/router/routes.js [hunk 2]
```diff
@@ -118,8 +118,8 @@ module.exports = [
 		}
 	},
 	{
-		path: '/checkout/eco-challenge/thanks',
-		component: () => import('@/pages/Thanks/ThanksPageEcoChallenge'),
+		path: '/checkout/iwd-challenge/thanks',
```
Comment (dyersituations) [inline src/router/routes.js:121]: Do we need to redirect the old path or is it not linked to from anywhere?

**Comment by dyersituations** [comment]
Do we also want to remove the `EcoChallengeLightbox` + stories, or will that be re-used?

File: src/router/routes.js (hunk 2 above)
Comment (eddieferrer) [reply src/router/routes.js:121]: Its not linked.

**Comment by eddieferrer** [comment]
> Do we also want to remove the `EcoChallengeLightbox` + stories, or will that be re-used?

I think the lightbox is useful to keep around, it might be used in a future challenge

File: src/pages/Thanks/ThanksPageIwdChallenge.vue [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+		};
+
+		// The default empty object and the v-if will prevent the
+		// receipt from rendering in the rare cases this query fails.
+		// But it will not throw a server error.
+		this.receipt = data?.shop?.receipt ?? null;
+
+		let achievementData = {};
+		try {
+			achievementData = this.apollo.readQuery({
```
[the remark this answers is not in the document]
Comment (eddieferrer) [reply src/pages/Thanks/ThanksPageIwdChallenge.vue:254]: The achievement service gets notified of the loan purchase event and adjusts accordingly. More info here: https://github.com/kiva/achievements-service

**Review by dyersituations — APPROVED** [review]

**Review by mcstover — APPROVED** [review]

**Pull request merged**